src/Microsoft.Data/DataFrameColumn.cs

```diff
@@ -108,12 +108,12 @@ public void SetName(string newName, DataFrame dataFrame = null)
         public virtual DataFrameColumn FillNulls(object value, bool inPlace = false) => throw new NotImplementedException();
 
         // Arrow related APIs
-        protected internal virtual Field Field() => throw new NotImplementedException();
+        protected internal virtual Field GetArrowField() => throw new NotImplementedException();
         /// <summary>
         /// Returns the max number of values that are contiguous in memory
         /// </summary>
-        protected internal virtual int MaxRecordBatchLength(long startIndex) => 0;
-        protected internal virtual Apache.Arrow.Array AsArrowArray(long startIndex, int numberOfRows) => throw new NotImplementedException();
+        protected internal virtual int GetMaxArrowRecordBatchLength(long startIndex) => 0;
```

Comment:
Does this really need to be prefixed with `Arrow`? It could be generally useful for other things, not?

Reply:
Potentially. Nobody else is using it now, but you're right that it doesn't need the `Arrow` moniker.